test/handshake_helper.c

```diff
@@ -401,13 +401,131 @@ static void configure_handshake_ssl(SSL *server, SSL *client,
                                  ssl_servername_name(extra->client.servername));
 }
 
-
+/* The status for each connection phase. */
 typedef enum {
     PEER_SUCCESS,
     PEER_RETRY,
     PEER_ERROR
 } peer_status_t;
 
+/* An SSL object and associated read-write buffers. */
+typedef struct {
+    SSL *ssl;
+    /* Buffer lengths are int to match the SSL read/write API. */
+    unsigned char *write_buf;
+    int write_buf_len;
+    unsigned char *read_buf;
+    int read_buf_len;
+    int bytes_to_write;
+    int bytes_to_read;
+    peer_status_t status;
+} PEER;
+
+static void create_peer(PEER *peer, SSL_CTX *ctx)
+{
+    static const int peer_buffer_size = 64 * 1024;
```

Comment:
Nope, that's just the amount of data to send in one go. I made this up (the old tests use 40kB) and will tweak this after I understand how the multi-buffer tests work.